js/bundle.go

```diff
@@ -335,6 +336,20 @@ func (b *Bundle) instantiate(init *InitContext, vuID uint64) (err error) {
 	init.moduleVUImpl.eventLoop = eventloop.New(init.moduleVUImpl)
 	pgm := b.initializeProgramObject(rt, init)
 
+	// TODO: make something cleaner for interrupting scripts, and more unified
+	// (e.g. as a part of the event loop or RunWithPanicCatching()?
+	initCtxDone := init.moduleVUImpl.ctx.Done()
+	initDone := make(chan struct{})
+	watchDone := make(chan struct{})
+	go func() {
+		select {
+		case <-initCtxDone:
+			rt.Interrupt(init.moduleVUImpl.ctx.Err())
```

Comment:
I need to check what the logic is in the other cases (not init) but if it is the same maybe could we delegate this listener directly in moduleVUImpl? :thinking: 

If the logic is different, we should use this opportunity to wrap the error so we know it is generated by the init process.

Reply:
Unfortunately it wasn't straightforward to unify this init interrupt logic with the rest of the test abort logic during test execution :disappointed: At least I didn't see how to do that without a much, much  bigger refactoring of the whole `js/` package... :disappointed: 

So I chose to make only minor improvements in this PR (see the [`Simplify the js/Bundle.instantiate() API`](https://github.com/grafana/k6/pull/2800/commits/e076d4001dcd65d4b0473b98f07a7662eee9a47b) commit) and leave the rest for a future PR (see the TODO in the code snippet you've selected).

Reply:
>If the logic is different, we should use this opportunity to wrap the error so we know it is generated by the init process.

What about this, is it doable?

Reply:
Hmm not sure what additional information we can add here, as the rest of the code is now :thinking: I think it will be much easier to fix this after we have the `NewTestRunContext()` in a "higher" place, i.e. after https://github.com/grafana/k6/pull/2813 :thinking: